src/Channel.cpp

```diff
@@ -84,7 +87,85 @@ void Channel::remove(Channel *c) {
 	QWriteLocker lock(&c_qrwlChannels);
 	c_qhChannels.remove(c->iId);
 }
-#endif
+
+void Channel::setFilterMode(Channel::FilterMode filterMode) {
+	m_filterMode = filterMode;
+	ServerHandlerPtr sh = Global::get().sh;
+	if (sh) {
+		Global::get().db->setChannelFilterMode(sh->qbaDigest, iId, m_filterMode);
+	}
+}
+
+bool Channel::isFiltered() const {
+	if (!Global::get().s.bFilterActive) {
+		// Channel filtering is disabled
+		return false;
+	}
+
+	Channel *userChannel = nullptr;
```

Comment:
```suggestion
	const Channel *userChannel = nullptr;
```
From a quick glance the channel object is not changed in this function and can therefore be labelled `const`